src/main/java/org/xstefank/model/PersistentList.java

```diff
@@ -0,0 +1,74 @@
+package org.xstefank.model;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class PersistentList {
+
+    private List<String> list = new ArrayList<>();
+    private File file;
+
+    public PersistentList(String dirName, String fileName) {
+        if (dirName == null && fileName == null) {
```

Comment:
this will allow to `dirName == null` and `fileName != null` which can cause NPE later